wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/ParallelDeadlineGroup.java

```diff
@@ -52,87 +41,15 @@ public ParallelDeadlineGroup(Command deadline, Command... commands) {
    * @param deadline the command that determines when the group ends
    */
   public void setDeadline(Command deadline) {
-    if (!m_commands.containsKey(deadline)) {
+    if (!getCommandStatuses().containsKey(deadline)) {
       addCommands(deadline);
     }
     m_deadline = deadline;
   }
 
-  /**
-   * Adds the given commands to the group.
-   *
-   * @param commands Commands to add to the group.
-   */
-  public final void addCommands(Command... commands) {
-    if (!m_finished) {
-      throw new IllegalStateException(
-          "Commands cannot be added to a composition while it's running");
-    }
-
-    CommandScheduler.getInstance().registerComposedCommands(commands);
-
-    for (Command command : commands) {
-      if (!Collections.disjoint(command.getRequirements(), m_requirements)) {
-        throw new IllegalArgumentException(
-            "Multiple commands in a parallel group cannot" + "require the same subsystems");
-      }
-      m_commands.put(command, false);
-      m_requirements.addAll(command.getRequirements());
-      m_runWhenDisabled &= command.runsWhenDisabled();
-      if (command.getInterruptionBehavior() == InterruptionBehavior.kCancelSelf) {
-        m_interruptBehavior = InterruptionBehavior.kCancelSelf;
-      }
-    }
-  }
-
-  @Override
-  public final void initialize() {
-    for (Map.Entry<Command, Boolean> commandRunning : m_commands.entrySet()) {
-      commandRunning.getKey().initialize();
-      commandRunning.setValue(true);
-    }
-    m_finished = false;
-  }
-
-  @Override
-  public final void execute() {
-    for (Map.Entry<Command, Boolean> commandRunning : m_commands.entrySet()) {
-      if (!commandRunning.getValue()) {
-        continue;
-      }
-      commandRunning.getKey().execute();
-      if (commandRunning.getKey().isFinished()) {
-        commandRunning.getKey().end(false);
-        commandRunning.setValue(false);
-        if (commandRunning.getKey().equals(m_deadline)) {
-          m_finished = true;
-        }
-      }
-    }
-  }
-
-  @Override
-  public final void end(boolean interrupted) {
-    for (Map.Entry<Command, Boolean> commandRunning : m_commands.entrySet()) {
-      if (commandRunning.getValue()) {
-        commandRunning.getKey().end(true);
-      }
-    }
-  }
-
   @Override
   public final boolean isFinished() {
-    return m_finished;
-  }
-
-  @Override
-  public boolean runsWhenDisabled() {
-    return m_runWhenDisabled;
-  }
-
-  @Override
-  public InterruptionBehavior getInterruptionBehavior() {
-    return m_interruptBehavior;
+    return m_deadline.isFinished();
```

Comment:
This will be a second call to `isFinished` in the same loop.
Query the map instead.